presto-spi/src/main/java/com/facebook/presto/spi/statistics/HistoricalPlanStatistics.java

```diff
@@ -16,34 +16,70 @@
 import com.facebook.drift.annotations.ThriftConstructor;
 import com.facebook.drift.annotations.ThriftField;
 import com.facebook.drift.annotations.ThriftStruct;
-import com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 
-import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
 @ThriftStruct
 public class HistoricalPlanStatistics
 {
-    // TODO: Add more data for historical runs, like avg/min/max over last N days/runs
-    private final PlanStatistics lastRunStatistics;
+    private static final HistoricalPlanStatistics EMPTY = new HistoricalPlanStatistics(Optional.empty(), Optional.empty(), Optional.empty());
+    // Thrift values are stored as optional, to make them easier to change in future
+
+    // When this field is present, only these statistics are used.
+    private final Optional<PlanStatistics> overriddenStatistics;
+    // Plan statistics from last `LAST_RUNS` runs
+    private final Optional<List<PlanStatistics>> lastRunsStatistics;
+    // Input table statistics from last `LAST_RUNS` runs
+    private final Optional<List<List<PlanStatistics>>> lastRunsInputStatistics;
```

Comment:
Updated comment.

Values are optional for thrift backwards compatibility, so when we change this structure we can simply drop these fields, instead of setting them to an empty list. This is documented by a comment above these fields